custom_connectors/custom_auth/trackvia_connector.rb

```diff
@@ -0,0 +1,896 @@
+            type: 'number',
+            control_type: 'number',
+            optional: false
+          },
+          {
+            name: 'longitude',
+            type: 'number',
+            control_type: 'number',
+            optional: false
+          }
+        ]
+      end
+    end,
+
+    get_delimeter: lambda do |input|
+      type = input[:type]
+      ',' if type == 'checkbox'
+    end,
+
+    get_toggle_hint: lambda do |input|
+      type = input[:type]
+      if type == 'dropDown'
+        'Select an option from the list'
+      elsif type == 'checkbox'
+        'Select one or more options from the list'
+      end
+    end,
+
+    get_toggle_field: lambda do |input|
+      type = input[:type]
+      name = input[:name]
+      required = input[:required]
+      if %w[dropDown checkbox].include?(type)
+        {
+          name: name,
+          label: name,
+          type: :string,
+          control_type: 'text',
+          optional: required,
+          toggle_hint: "Set #{name} manually",
+          hint: 'Please ensure that the value provided is valid'
+        }
+      end
+    end,
+
+    get_output_fields: lambda do |input|
+      view_id = input[:view_id]
+      get("/openapi/views/#{view_id}")['structure']
+        .map do |field|
+        {
+          name: field['name'],
+          label: field['name'],
+          optional: !field['required'],
+          type: call(:get_type,
+                     type: field['type'],
+                     name: field['name']),
+          control_type: call(:get_control_type,
+                             type: field['type']),
+          properties: call(:get_properties,
+                           type: field['type'])
+        }
+      end
+    end,
+
+    get_fields: lambda do |input|
+      view_id = input[:view_id]
+      structure = get("/openapi/views/#{view_id}")['structure']
+                  .reject { |field| !field['canCreate'] || !field['canUpdate'] }
+      structure.map do |field|
+        {
+          name: field['name'],
+          label: field['name'],
+          optional: !field['required'],
+          type: call(:get_type, type: field['type']),
+          control_type: call(:get_control_type,
+                             type: field['type'],
+                             name: field['name']),
+          pick_list: call(:get_picklist_options,
+                          choices: field['choices']),
+          delimiter: call(:get_delimeter,
+                          type: field['type']),
+          properties: call(:get_properties,
+                           type: field['type']),
+          toggle_hint: call(:get_toggle_hint,
+                            type: field['type'],
+                            name: field['name']),
+          toggle_field: call(:get_toggle_field,
+                             type: field['type'],
+                             name: field['name'],
+                             required: !field['required'])
+        }
```

Comment:
@skennicutt Use `compact` for safe hash, make sure you have no properties with nil
e.g.
```
hash = { a: true, b: false, c: nil}
hash.compact # => { a: true, b: false}
```
URL: https://apidock.com/rails/Hash/compact